src/shared/notify-recv.c

```diff
@@ -80,6 +85,24 @@ int notify_recv(
                 }
         }
 
+        _cleanup_(fdset_free_asyncp) FDSet *fds = NULL;
+
+        if (n_fds > 0) {
+                assert(fd_array);
+
+                if (!ret_fds) {
+                        log_debug("Received fds via notification while none is expected, closing all.");
+                        close_many(fd_array, n_fds);
```

Comment:
hmm, maybe we should use async closing here too... after all people might send us weird shit.

you are using fdset_free_asyncp above after all...

Reply:
I wonder if asynchronous_close() should be taught about close_range in this case. Seems rather painful to spawn a whole flow of processes otherwise...